src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_validators.py

```diff
@@ -33,6 +38,20 @@ def validate_parent(string):
     ParentType = collections.namedtuple('ParentType', 'type name')
     return ParentType(type=type_comp, name=name_comp)
 
+def validate_deployment_name(namespace):
+    #If missing,try come out with a name associated with the template name
+    if namespace.deployment_name is None:
+        #if we get json content, default to a fixed name
+        if namespace.template_file_path[:1] == '{':
+            namespace.deployment_name = 'deployment1'
```

Comment:
I think this will potentially result in collisions given that deployments don't get deleted in a RG. Burt has run into this issue many times writing smart creates. I'd recommend you append some kind of random string at the end instead of a hard coded value like that.